Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -2187,6 +2187,79 @@ extension EventLoopFuture {
     }
 }
 
+// MARK: assertion
+
+extension EventLoopFuture {
+    /// Returns a new `EventLoopFuture` that asserts the original future's success.
+    ///
+    /// If the original future fails, it triggers an assertion failure, causing a runtime error during development.
+    @inlinable
+    public func assertSuccess() -> EventLoopFuture<Value> {
+        let promise = self.eventLoop.makePromise(of: Value.self)
```

Comment:
No need to create a promise in any of these functions; you can use `always` instead:

```swift
return self.always { result in 
    switch result { 
    case .success:
        ()
    case .failure(let error):
        assertionFailure("Expected success, but got failure: \(error)")    
    }
}
```

Reply:
Indeed, but we should pass `file` and `line` to `assertionFailure` such that it shows where the user put `.assertX` and not always this line

Reply:
oh okay i see no need to create a new promise / or complete it bc `always` can take care of that for us, working on the changes. Thank you!

Reply:
changes commited!